### PR TITLE
[move-2024] Migrate examples to move2024

### PR DESCRIPTION
**Pull request by 0xaslan**
## Description 

Updates the example packages to use Move 2024 edition. Uses the latest migration tool. All examples successfully migrated, built, and unit tested.

The receiver syntax changes will be in a separate PR.

## Test Plan 

The unit tests all pass with the latest changes.

---
If your changes are not user-facing and do not break anything, you can skip the following section. Otherwise, please briefly describe what has changed under the Release Notes section.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

**Review by leecchh — APPROVED** [review]

File: examples/move/bridge/sources/treasury.move [hunk 1]
```diff
@@ -14,44 +14,44 @@ module bridge::treasury {
...
+    public(package) fun create(ctx: &mut TxContext): BridgeTreasury {
         // assert!(tx_context::sender(ctx) == @0x0, ENotSystemAddress);
         BridgeTreasury {
             treasuries: object_bag::new(ctx)
         }
     }
 
-    public(friend) fun burn<T>(self: &mut BridgeTreasury, token: Coin<T>, ctx: &TxContext) {
+    public(package) fun burn<T>(self: &mut BridgeTreasury, token: Coin<T>, ctx: &TxContext) {
     // public(friend) fun burn<T>(self: &mut BridgeTreasury, token: Coin<T>, ctx: &mut TxContext) {
```
Comment (tzakian) [inline examples/move/bridge/sources/treasury.move:39]: Not related to this PR I think, but somewhat confused by the commented-out function signature below this one? Might be best to remove unless there's a good reason to keep it?

File: examples/move/bridge/sources/treasury.move [hunk 2]
```diff
@@ -14,44 +14,44 @@ module bridge::treasury {
...
     }
 
-    public(friend) fun mint<T>(self: &mut BridgeTreasury, amount: u64, ctx: &mut TxContext): Coin<T> {
+    public(package) fun mint<T>(self: &mut BridgeTreasury, amount: u64, ctx: &mut TxContext): Coin<T> {
         create_treasury_if_not_exist<T>(self, ctx);
         let treasury = object_bag::borrow_mut(&mut self.treasuries, type_name::get<T>());
         coin::mint(treasury, amount, ctx)
     }
 
     // fun create_treasury_if_not_exist<T>(self: &mut BridgeTreasury, ctx: &mut TxContext) {
```
Comment (tzakian) [inline examples/move/bridge/sources/treasury.move:51]: Same question here as above about the commented-out function signature.

File: examples/move/bridge/sources/usdc.move [hunk 3]
```diff
@@ -8,9 +8,9 @@ module bridge::usdc {
     use sui::transfer;
     use sui::tx_context::{Self, TxContext};
 
-    friend bridge::treasury;
+    /* friend bridge::treasury; */
```
Comment (tzakian) [inline examples/move/bridge/sources/usdc.move:11]: can remove

File: examples/move/bridge/sources/usdt.move [hunk 4]
```diff
@@ -8,9 +8,9 @@ module bridge::usdt {
     use sui::transfer;
     use sui::tx_context::{Self, TxContext};
 
-    friend bridge::treasury;
+    /* friend bridge::treasury; */
```
Comment (tzakian) [inline examples/move/bridge/sources/usdt.move:11]: ditto

File: examples/move/token/sources/regulated_token.move [hunk 5]
```diff
@@ -109,7 +109,7 @@ module examples::regulated_token {
         treasury_cap
     }
 
-    #[test_only] friend examples::regulated_token_tests;
+    /* #[test_only] */ /* friend examples::regulated_token_tests; */
```
Comment (tzakian) [inline examples/move/token/sources/regulated_token.move:112]: Can remove

File: examples/move/token/sources/simple_token.move [hunk 6]
```diff
@@ -65,7 +65,7 @@ module examples::simple_token {
         treasury_cap
     }
 
-    #[test_only] friend examples::simple_token_tests;
+    /* #[test_only] */ /* friend examples::simple_token_tests; */
```
Comment (tzakian) [inline examples/move/token/sources/simple_token.move:68]: ditto

**Review by tzakian — APPROVED** [review]
Overall LGTM! Just a couple nits around comments that can probably be removed now.

File: examples/move/bridge/sources/btc.move [hunk 7]
```diff
@@ -8,9 +8,9 @@ module bridge::btc {
     use sui::transfer;
     use sui::tx_context::{Self, TxContext};
 
-    friend bridge::treasury;
+    /* friend bridge::treasury; */
```
[the remark this answers is not in the document]
Comment (0xaslan) [reply examples/move/bridge/sources/btc.move:11]: removed all of the friend comments

File: examples/move/bridge/sources/treasury.move (hunk 1 above)
Comment (0xaslan) [reply examples/move/bridge/sources/treasury.move:39]: I don't see a good reason to have it, removing

**Pull request merged**